internal_ws/ykcompile/src/lib.rs

```diff
@@ -408,7 +410,7 @@ impl Location {
 
     /// Apply an offset to the location, returning a new one.
     fn offset(self, off: OffT) -> Self {
-        if off == 0 {
+        if off == 0 { // Is the special case a performance hack or ... ?
```

Comment:
It's not strictly necessary, unless removing it causes us to hit the panic case. But yes, it is a small optimisation too.

Reply:
I'm actually slightly worried that it might not be correct, or at least causes a surprise: if I pass `Location::Reg` and `off != 0` I get a panic, but with `off == 0` I don't. I'm not even sure what it means to offset a register.

Reply:
I actually assumed this would be `unreachable` for registers and `offset != 0`. At least I've not seen a register/constant being offset, so we need to investigate this.

Reply:
> I'm not even sure what it means to offset a register.

This is currently undefined, so it probably should crash.

Reply:
Right, so I think we should remove the `if` -- the optimisation probably isn't significant, and it potentially hides some bad cases.

Reply:
Fixed in 2c47787fbaaf271fdb2fef39104bbf681eee6454.